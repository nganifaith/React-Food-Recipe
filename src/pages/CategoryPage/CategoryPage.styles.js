import styled from 'styled-components';

export const Wrapper = styled.div``;
export const Category = styled.div``;
export const Content = styled.div``;
export const MealImage = styled.img``;
