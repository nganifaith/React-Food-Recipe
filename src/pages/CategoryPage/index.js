import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Categories from '../../components/MealCategory';
import MealImage from '../../components/MealImage';
import Main from '../../components/Main';

import { getMealsByCategory } from '../../API';

import { Wrapper, Content, Category } from './CategoryPage.styles';

import { getMeals } from '../../reducers/mealsList';
import { setMeals } from '../../actions';

const CategoryPage = () => {
  const [category, setCategory] = useState('Pasta');
  const [error, setError] = useState('');
  const dispatch = useDispatch();
  const meals = useSelector(getMeals);

  useEffect(() => {
    setError('');
    getMealsByCategory(category)
      .then(({ meals }) => {
        dispatch(setMeals(meals));
      })
      .catch(() => {
        setError('Error fetching Data');
      });
  }, [category]);

  return (
    <Wrapper>
      <Content>
        <Category>
          <Categories current={category} onCategorySelected={setCategory} />
          <span>{error}</span>
        </Category>
        <Main header={`Category ~ ${category}`}>
          {meals.map((meal) => (
            <MealImage key={meal.idMeal} meals={meal} />
          ))}
        </Main>
      </Content>
    </Wrapper>
  );
};

export default CategoryPage;
