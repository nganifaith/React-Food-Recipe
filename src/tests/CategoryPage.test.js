import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from '../reducers';

import CategoryPage from '../pages/CategoryPage';

const store = createStore(rootReducer);

it('renders correctly', () => {
  const home = renderer
    .create(
      <Provider store={store}>
        <CategoryPage />
      </Provider>
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});
