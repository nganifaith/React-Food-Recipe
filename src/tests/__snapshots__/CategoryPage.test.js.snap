// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gKseQn"
>
  <div
    className="sc-fubCzh"
  >
    <div
      className="sc-iBPTik"
    >
      <div
        className="sc-bdfBQB"
      >
        <div
          className="sc-gsTEea dPzzKC"
        />
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="sc-eCstlR iYFpZf"
    >
      <h1>
        Category ~ Pasta
      </h1>
      <div
        className="sc-jSgvzq ehCKSs"
      />
    </div>
  </div>
</div>
`;
